### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by Dianap19d**

### CODE REVIEW

File: reports/README.md [hunk 1]
```diff
@@ -182,7 +196,7 @@ Note: This project (currently) uses Google Drive as the storage location for DVC
 >
 > Answer:
 
---- question 6 fill here ---
+---  We implemented rules for code quality and formatting through pre-commit hooks, using ruff for linting and auto-fixing issues. In large projects, consistent code quality and format are vital as they lead to a more maintainable and understandable codebase, reduce the risk of introducing errors, and make collaboration easier. Uniform code standards help ensure that any developer can quickly read and contribute to the code, regardless of the project's scale.---
```
Comment (truszkowski-tomasz) [inline reports/README.md]:
```suggestion
---  We implemented rules for code quality and formatting through pre-commit configuration. We are using standard pre-commit hooks:
      - id: end-of-file-fixer
      - id: trailing-whitespace
      - id: check-docstring-first
      - id: check-executables-have-shebangs
      - id: check-case-conflict
      - id: detect-private-key

Also, in order to follow `pip-8` standard, we are also using ruff hooks for linting and code formating.
In large projects, consistent code quality and format are vital as they lead to a more maintainable and understandable codebase, reduce the risk of introducing errors, and make collaboration easier. Uniform code standards help ensure that any developer can quickly read and contribute to the code, regardless of the project's scale.---
```

File: reports/README.md [hunk 2]
```diff
@@ -262,7 +276,13 @@ Note: This project (currently) uses Google Drive as the storage location for DVC
 >
 > Answer:
 
---- question 11 fill here ---
+--- Our continuous integration (CI) setup is configured within our GitHub Actions workflow to ensure that every change to the codebase is automatically tested and validated. The CI pipeline includes running unit tests, linting, and other code quality checks.
+
+The unit tests are executed using a testing framework such as pytest to validate that all components of our application behave as expected. Linting is performed to enforce coding standards and style consistency across the project, which could be done using tools like  ruff.
```
Comment (truszkowski-tomasz) [inline reports/README.md]:
```suggestion
--- Our continuous integration (CI) setup is configured within GitHub Actions workflows to ensure that every change to the codebase is automatically tested and validated. Github Actions consist of two workflows - first one executes all the unittests using pytest framework to validate that all components of our application behave as expected. Second, using ruff, enforce coding standards and style consistency across the project.  
```

File: reports/README.md [hunk 3]
```diff
@@ -296,7 +316,11 @@ Note: This project (currently) uses Google Drive as the storage location for DVC
 >
 > Answer:
 
---- question 13 fill here ---
+--- To ensure reproducibility in our experiments, we employed a combination of config files and careful environment management. Whenever an experiment is run, we use YAML configuration files that specify all experiment parameters. This setup allows for precise control over the experiment settings and easy adjustment without modifying the code. Each experiment's configuration is saved alongside its results, providing a clear record of the conditions under which it was run.
```
Comment (truszkowski-tomasz) [inline reports/README.md]:
```suggestion
--- To ensure reproducibility in our experiments, we employed a combination of config files, and environment management.. Whenever an experiment is run, we use hydra framework configuration files that specify all parameters. This setup allows for precise control over the experiment settings and easy adjustment without modifying the code. Each experiment's configuration is saved alongside its results, providing a clear record of the conditions under which it was run.
```

File: reports/README.md [hunk 4]
```diff
@@ -171,7 +171,21 @@ Note: This project (currently) uses Google Drive as the storage location for DVC
...
+- `reports/`: Stores generated analysis reports or results related to the project.
+- `dockerfiles/`: Contains Dockerfiles which define the environment for running the project in containers.
+- `.github/workflows/`: Provides the configuration for GitHub Actions to automate the testing and deployment pipeline.
+- `notebooks/`: Intended for Jupyter notebooks, currently unused but available for future exploratory data analysis or interactive development.
+
+Additionally, the root of the repository includes several configuration files and directories that aid in the development and deployment of the project:
+
+- `docker-compose.yaml`: For defining and running multi-container Docker applications.
+- `requirements.txt`: Lists all Python dependencies for the project.
+
```
Comment (truszkowski-tomasz) [inline reports/README.md:181]: This section should be changed to reflect structure in readme + few files/folders that have been added along the way.

File: reports/README.md [hunk 5]
```diff
@@ -366,7 +390,20 @@ Finally, as for memory usage, the sections that consume the most memory are the
...
+Cloud Build: Configured to automatically build a new container image for training whenever code was pushed to the main branch of our repository, ensuring that our training environment was always up-to-date.
+
+Container Registry: Stored the container images created by Cloud Build, serving as a private repository for our Docker images.
+
+Vertex AI: Utilized for submitting custom training jobs using the container images from Container Registry, which allowed us to train our models on scalable infrastructure.
+
+BigQuery: Used to store inputs and outputs of our deployed models, enabling us to analyze model performance and detect data drift over time.
+
+Compute Engine: Employed for initial testing and running experiments on the cloud before scaling up , providing customizable and on-demand virtual  ---
```
Comment (truszkowski-tomasz) [inline reports/README.md:407]: @notaiax @Shah-Bekhsh Could you guys take a look at this?

File: reports/README.md [hunk 6]
```diff
@@ -488,7 +526,18 @@ Finally, as for memory usage, the sections that consume the most memory are the
 >
 > Answer:
 
---- question 26 fill here ---
+--- 
+One of the major struggles in our project was the integration and orchestration of various cloud services and tools to establish a smooth workflow. We spent a considerable amount of time setting up and configuring our CI/CD pipeline, to ensure that our machine learning models were automatically trained and evaluated upon each code push.
```
Comment (truszkowski-tomasz) [inline reports/README.md]: It's worth adding that we spent a looooot of time struggling with dvc usage inside the cloud and we end up with workaround.

**Review by truszkowski-tomasz — CHANGES_REQUESTED** [review]

File: reports/README.md (hunk 5 above)
Comment (Shah-Bekhsh) [reply reports/README.md:407]: Please remove BigQuery from this list as we have not used it in our project.

File: reports/README.md [hunk 7]
```diff
@@ -381,7 +425,8 @@ Finally, as for memory usage, the sections that consume the most memory are the
 >
 > Answer:
 
---- question 18 fill here ---
+--- While we had the capability to train our model locally using our CPUs, the necessity for scaling to more intensive compute resources was not immediate. Consequently, we didn't fully explore the potential of Compute Engine for training purposes. For model deployment, hosting our FastAPI server on a Compute Engine VM was a possibility. Yet, we recognized that this option might incur higher costs and offer less scalability compared to serverless alternatives like Cloud Run, which would be our preferred choice for such needs. Cloud Run provides a managed environment that automatically scales based on traffic, making it a more cost-effective and flexible solution for deploying containerized applications like ours.
+ ---
```
Comment (Shah-Bekhsh) [inline reports/README.md:423]: @notaiax  can you talk a little bit about the VM you used to train the model here? It would be very useful.

**Review by Shah-Bekhsh — APPROVED** [review]

**Review by Shah-Bekhsh — APPROVED** [review]

**Review by Shah-Bekhsh — APPROVED** [review]

**Pull request merged**